portal/v2/src/ClusterDetailListComponents/Statistics/GraphOptionsComponent.tsx

```diff
@@ -0,0 +1,220 @@
+      let localDate: Date = new Date();
+      localDate.setUTCFullYear(date!.getFullYear(), date!.getMonth(), date!.getDate())
+      localDate.setHours(props.endDate.getHours());
+      localDate.setMinutes(props.endDate.getMinutes());
+      props.setEndDate(localDate)  
+     };
+     
+    return (
+      <>
+      <Stack horizontal verticalAlign="center">
+      <Stack horizontal verticalAlign="center" className={classNames.iconContainer} /* style={{ boxShadow: theme.effects.elevation8 }}*/>
+          <TooltipHost>
+            <IconButton
+              onClick={_decreaseDuration}
+              iconProps={calculatorSubtract}
+              styles={iconButtonStyles}
+            />
+          </TooltipHost>
+          <TooltipHost>
+            <Text styles={durationStyle}>{props.duration}</Text>
+          </TooltipHost>
+          <TooltipHost >
+            <IconButton
+              onClick={_increaseDuration}
+              iconProps={calculatorAddition}
+              styles={iconButtonStyles}
+            />
+          </TooltipHost>
+        </Stack>
+        <Stack horizontal verticalAlign="center" className={dateTimePickerStyles.iconContainer}  /* style={{ boxShadow: theme.effects.elevation8 }}*/>
+          <DatePicker
+            styles={datePickerStyles}
+            placeholder="End Date"
+            ariaLabel="End Date"
+            onSelectDate={onDateChange}
+            value={convertToUTC(props.endDate)}
+            allowTextInput
+          />
+          <TimePicker
+            styles={timePickerStyles}
+            allowFreeform
+            placeholder={timeToString(convertToUTC(props.endDate))}
+            autoComplete="on"
+            onChange={onTimeChange}
+            defaultValue={convertToUTC(props.endDate)}
+            useComboBoxAsMenuWidth
+          />
+        </Stack>
+      </Stack>
+      </>
+    )
+  }
+
+  function timeToString(date: Date): string {
+    var str
+    let hourString = date.getHours().toString()
+    if (hourString.length === 1) {
+      str = "0" + hourString + ":"
+    } else {
+      str = hourString + ":"
+    }
+    let minuteString = date.getMinutes().toString()
+    if (minuteString.length === 1) {
+      str = str + "0" + minuteString
+    } else {
+      str += minuteString
+    }
+
+    return str
+  }
+
+  export const convertToUTC = (date: Date): Date => {
+    let localDate = new Date()
+    localDate.setFullYear(date.getUTCFullYear(), date.getUTCMonth(), date.getUTCDate())
+    localDate.setHours(date.getUTCHours())
+    localDate.setMinutes(date.getUTCMinutes())
+    
+    return localDate
+  }
+
+  export const convertTimeToHours = (duration: string): string => {
+    switch(duration) {
+      case "1d":
+        return "24h"
+      case "2d":
+        return "48h"
+      case "1w":
+        return "168h"
+      case "2w":
+        return "336h"
+      case "4w":
+        return "672h"
+      case "8w":
+        return "1344h"
+    }
+    return duration
+  }
```

Comment:
This entire function should just be a js object constant.

Reply:
I would like to keep it as a function only .. but I will replace the switch case with a js object.